src/ogc/wcs/WebCoverageService.js

```diff
@@ -35,27 +35,67 @@ define([
         /**
          * Represents a Web Coverage Service and provides functionality for interacting with the service. Includes
          * functionality for retrieving DescribeCoverage documents and providing WCS version agnostic coverage objects.
-         * @param serviceAddress the url of the Web Coverage Service
          * @constructor
          */
-        var WebCoverageService = function (serviceAddress) {
+        var WebCoverageService = function () {
+
+            /**
+             * The URL for the Web Coverage Service
+             */
+            this.serviceAddress = null;
 
             /**
              * A collection of the coverages available from this service. Not populated until service is initialized by
              * the connect method.
              * @type {Array}
              */
             this.coverages = [];
+
+            /**
+             * The WCS GetCapabilities document for this service.
+             * @type {WcsCapabilities}
+             */
+            this.capabilities = null;
+
+            /**
+             * A map of the coverages to their corresponding DescribeCoverage documents.
+             * @type {{}}
+             */
+            this.coverageDescriptions = {};
         };
 
         /**
-         * Connects to the Web Coverage Service specified in the constructor. This function handles version negotiation
-         * and capabilities document retrieval. The return is a Promise which returns the initialized
-         * WebCoverageService.
-         * @returns {Promise} a Promise of a WebCoverageService
+         * An array of compatible WCS versions supported by this client service.
+         * @type {string[]}
+         */
+        WebCoverageService.COMPATIBLE_WCS_VERSIONS = ["1.0.0", "2.0.0", "2.0.1"];
+
+        /**
+         * The maximum length of a constructed DescribeCoverage url request.
+         * @type {number}
+         */
+        WebCoverageService.MAX_URL_LENGTH = 2083;
```

Comment:
This value is somewhat magical. I remembered the WMS issue with too long URLs. I did a quick search and found IE doesn't allow urls longer than 2083 characters. I assumed this was the limitation but after finding the original issue, I found it was actually an Apache limit of 4096. I'm not even sure the 2083 applies to XHR requests in IE. I was ready to change to 4096 but noticed one of my test servers (http://sedac.ciesin.columbia.edu/geoserver/wcs) fails with longer URLs. It works with urls less than 2083 but not with 4096. I don't know where the cutoff is. I retained the maximum however because I believe it is a good practice for larger WCS servers.